hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/HandleContext.java

```diff
@@ -58,4 +59,16 @@ public interface HandleContext {
      * @return the validator for expiry metadata
      */
     ExpiryValidator expiryValidator();
+
+    /**
+     * Create a new store given the store's interface. This gives read-only access to the store.
+     *
+     * @param storeInterface The store interface to find and create a store for
+     * @return An implementation of store interface provided, or null if the store
+     * @param <C> Interface class for a Store
+     * @throws IllegalArgumentException if the storeInterface class provided is unknown to the app
+     * @throws NullPointerException if {@code clazz} is {@code null}
+     */
+    @NonNull
+    <C> C createStore(@NonNull Class<C> storeInterface);
```

Comment:
I think we should name this method `createReadableStore()` to avoid any confusion with writable stores.